src/requests/typings/function.ts

```diff
@@ -158,9 +160,21 @@ export type AppEventTransformationResponse = {
 
 export type AppEventHandlerResponse = void
 
-export type AppActionRequest = {
+/**
+ * The app action request body will contain different parameters depending on the category of the app action
+ *
+ * Specify your app action category as the generic type `Category` to get the correct body type,
+ * e.g. `const { body: { message, recipient }} = event as AppActionRequest<'Notification.v1.0'>`
+ *
+ * If you are using the Custom category, you can specify the parameter shape as the second generic type `CustomCategoryBody`,
+ * e.g. `const { body: { myNumber }} = event as AppActionRequest<'Custom', { myNumber: number }>`
+ */
+export type AppActionRequest<
+  CategoryType extends AppActionCategoryType = 'Custom',
+  CustomCategoryBody = AppActionCategoryBodyMap['Custom'],
+> = {
   headers: Record<string, string | number>
-  body: Record<string, unknown>
+  body: CategoryType extends 'Custom' ? CustomCategoryBody : AppActionRequestBody<CategoryType>
   type: typeof APP_ACTION_CALL
 }
```

Comment:
```
export type AppActionRequest<CategoryType extends AppActionCategoryType = 'Custom'> = {
  headers: Record<string, string | number>
  body: AppActionRequestBody<CategoryType>
  type: typeof APP_ACTION_CALL
}
```
that AppActionBody utility type should do the job right? just so we don't have an inline ternary in the type here